mm2src/mm2_main/src/lp_swap/taker_swap_v2.rs

```diff
@@ -483,76 +865,125 @@ impl<MakerCoin: MmCoin, TakerCoin: MmCoin + SwapOpsV2> State for TakerPaymentSen
 
         if let Err(e) = state_machine.maker_coin.wait_for_confirmations(input).compat().await {
```

Comment:
I think taker will have to wait for confirmation of maker payment before spending the funding transaction. Concurrent confirmations of both taker and maker payments were only viable before introducing immediate refund path for taker. I think you are aware of this but didn't get to change it yet :)
P.S. This is just a reminder for next sprints where confirmations need to be rechecked and probably changed, ref. next comment.

Reply:
I assumed that we can still keep concurrent confirmations, though it should be carefully reviewed with regard to possible maker payment cancellation/replacement with another tx. Added to the checklist and will add to the next sprint discussion points.